docs/markdown/snippets/python35-eol.md

```diff
@@ -0,0 +1,15 @@
+## Python 3.5 support will be dropped in the next release
+
+The final [Python 3.5 release was 3.5.10 in September](https://www.python.org/dev/peps/pep-0478/#id4).
+This release series is now End-of-Life (EOL). The only LTS distribution that
+still only ships Python 3.5 is Ubuntu 16.04, which will be
+[EOL in April 2021](https://ubuntu.com/about/release-cycle).
```

Comment:
This is unfortunately not true:
Debian 9 "Stretch" (oldstable) is supported up to June 30, 2022 [0] and ships with python 3.5 [1].

This is an actual issue for OpenWrt and its buildbots, which are running debian oldstable. My patch to drop support for python 3.5 because of meson was rejected for that reason [2].

Meson is popular and widely used by now, please don't be so eager to remove support for older distros. Especially CI boxes aren't always updated to the latest and greatest.

[0] https://wiki.debian.org/LTS
[1] https://packages.qa.debian.org/p/python3-defaults.html
[2] http://patchwork.ozlabs.org/project/openwrt/patch/20210216072108.26107-1-a.heider@gmail.com/

Reply:
Perhaps, but even so it remains true that python 3.5 itself is EOL.

Are you just making a point that the wording here should be tweaked? Do you want meson to reintroduce support for end of life versions of python?

(Why do you need to upgrade meson on Debian oldstable anyway? The entire distro base is incredibly outdated and frozen in amber.)

Reply:
TBH, I agree that python EOL does not matter, we are not a web service.  What matters is keeping a viable code base that uses new features from the language instead of piling hacks on top of hacks. So there is a balance to be found, and I think 3.6 is not unreasonable at this time. At some point it gets OpenWrt's responsability to upgrade their buildbots if they want to remain a viable project.

Reply:
It does matter only in the sense that once it's EOL it's fair game for people to say "we really need this new feature, let's bump the minimum version".